evap/student/tests/test_views.py

```diff
@@ -0,0 +1,94 @@
+from django.urls import reverse
+from model_mommy import mommy
+
+from evap.evaluation.models import UserProfile, Course, Questionnaire, Question, Contribution
+from evap.evaluation.tests.tools import WebTest, ViewTest
+
+
+class TestStudentIndexView(ViewTest):
+    test_users = ['student']
+    url = '/student/'
+
+    def setUp(self):
+        # View is only visible to users participating in at least one course.
+        user = mommy.make(UserProfile, username='student')
+        mommy.make(Course, participants=[user])
+
+
+@override_settings(INSTITUTION_EMAIL_DOMAINS=["example.com"])
+class TestVoteView(WebTest):
+    fixtures = ['minimal_test_data']
+
+    def test_complete_vote_usecase(self):
+        """
+            Submits a student vote, verifies that an error message is
+            displayed if not all rating questions have been answered and that all
+            given answers stay selected/filled and that the student cannot vote on
+            the course a second time.
+        """
+        page = self.get_assert_200("/student/vote/5", user="lazy.student")
+        form = page.forms["student-vote-form"]
+        form["question_17_2_3"] = "some text"
+        form["question_17_2_4"] = 1
+        form["question_17_2_5"] = 6
+        form["question_18_1_1"] = "some other text"
+        form["question_18_1_2"] = 1
+        form["question_19_1_1"] = "some more text"
+        form["question_19_1_2"] = 1
+        form["question_20_1_1"] = "and the last text"
+        response = form.submit()
+
+        self.assertIn("vote for all rating questions", response)
+        form = page.forms["student-vote-form"]
+        self.assertEqual(form["question_17_2_3"].value, "some text")
+        self.assertEqual(form["question_17_2_4"].value, "1")
+        self.assertEqual(form["question_17_2_5"].value, "6")
+        self.assertEqual(form["question_18_1_1"].value, "some other text")
+        self.assertEqual(form["question_18_1_2"].value, "1")
+        self.assertEqual(form["question_19_1_1"].value, "some more text")
+        self.assertEqual(form["question_19_1_2"].value, "1")
+        self.assertEqual(form["question_20_1_1"].value, "and the last text")
+        form["question_20_1_2"] = 1  # give missing answer
+        form.submit()
+
+        self.get_assert_403("/student/vote/5", user="lazy.student")
+
+    def test_simple_vote(self):
+        page = self.get_assert_200("/student/vote/5", user="lazy.student")
+        form = page.forms["student-vote-form"]
+        form["question_17_2_3"] = "some text"
+        form["question_17_2_4"] = 1
+        form["question_17_2_5"] = 6
+        form["question_18_1_1"] = "some other text"
+        form["question_18_1_2"] = 1
+        form["question_19_1_1"] = "some more text"
+        form["question_19_1_2"] = 1
+        form["question_20_1_1"] = "and the last text"
+        form["question_20_1_2"] = 1
+        form.submit()
+
+        self.get_assert_403("/student/vote/5", user="lazy.student")
```

Comment:
i think the test should either test: submitting the vote and asserting that the results have been saved, or assert that a 403 is thrown if a student already voted. right now it kinda does a bit of both.